[Feature] cluster key

justfortaste opened the pull request:
Support secondary clustering key in MyRocks

```   
 Support new syntax
      create table t1(c1 int priamry key, c2 int clustering, c3 int) engine=rocksdb;
      alter table t1 add clustering key idx1(c3);

 New clustering key row layout as follows, PK and SK has not changed
    1) CK (secondary cluster key)
      key: index_id + ck_cols + pk_cols
      value: [TTL] +  NULL-bitmap + unpack_info + other_cols

    2) PK (primary key)
      key: index_id + pk_cols
      value: [TTL] +  NULL-bitmap + unpack_info + other_cols

    3) SK (ordinary sencordary key)
      key: index_id + sk_cols + pk_cols
      value: [unpack_info (TTL is in unpack_info, TTL_FLAG)
```

justfortaste committed:
Add support for CLUSTERING key type at server layer

port from percona
https://github.com/percona/percona-server/commit/9faf43d99b6c6e5ea90a977eedd9c73e90391a8f

justfortaste committed:
rocksdb support for CLUSTERING key

laurynas-biveinis reviewed (commented):
The syntax and server parts of the patch appear to be based on the Percona Server TokuDB clustering key support, which appears to be a good idea from future compatibility point of view

justfortaste committed:
Fix add clustering index bug and clean ttl relative code

bugfix: when add clustring index, we also need call
  convert_ck_record_to_storage_formatto set values

clean ttl relative code, then ttl layout in index as follows

1) PK
  key: index_id + pk_cols
  value: [TTL] +  NULL-bitmap + unpack_info + other_cols

2) CK
  key: index_id + ck_cols + pk_cols
  value: [TTL] +  NULL-bitmap + unpack_info + other_cols

3) SK
  key: index_id + sk_cols + pk_cols
  value: unpack_info (TTL is in unpack_info, TTL_FLAG)
